include/nccl-headers/neuron/error.h

```diff
@@ -6,7 +6,7 @@
 #ifndef NCCL_HEADERS_NEURON_ERROR_H_
 #define NCCL_HEADERS_NEURON_ERROR_H_
 
-#ifdef _cplusplus
```

Comment:
If we're doing all this work, let's just define macros in "config_bottom.h", something like:

```
#undef BEGIN_C_DECLS
#undef END_C_DECLS
#ifdef __cplusplus
# define BEGIN_C_DECLS extern "C" {
# define END_C_DECLS }
#else
#define BEGIN_C_DECLS
#define END_C_DECLS 
#endif
```

Reply:
We can do this, but I think it's more idiomatic to just put it directly inline.

Reply:
I've seen both; advantage of the macros is that most indenters then don't try to indent the entire header file.

Reply:
I understand the argument that it would have been a one line change instead of a 30 line change if we were using macros here, but now that it's using the right token, this should never need to be touched again until we're deleting these lines to transition the API to C++, and I will gladly write you a .dir-locals.el and a .clang-format.yml for the interim that does the right thing here in terms of indentation.

Reply:
Fair; i disagree, but not enough to dig  in my heals here.